src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -744,7 +739,7 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
-    std::optional<winrt::TerminalApp::FilteredCommand> CommandPalette::_buildCommandLineCommand(std::wstring const& commandLine)
+    std::optional<winrt::TerminalApp::FilteredCommand> CommandPalette::_buildCommandLineCommand(std::wstring const& commandLine)  const
```

Comment:
You can prevent the unneeded, costly string copy with `.c_str()` by replacing this function with:
```cpp
std::optional<winrt::TerminalApp::FilteredCommand> CommandPalette::_buildCommandLineCommand(const winrt::hstring& commandLine) const
{
    if (commandLine.empty())
    {
        return std::nullopt;
    }

    auto commandLinePaletteItem{ winrt::make<winrt::TerminalApp::implementation::CommandLinePaletteItem>(commandLine) };
    return winrt::make<FilteredCommand>(commandLinePaletteItem);
}
```

This will fail compilation though which is why you need to put this into `CommandPalette::_dispatchCommandline`:
```cpp
const auto filteredCommand = command ? command : _buildCommandLineCommand(winrt::hstring{ _getTrimmedInput() });
```

Reply:
Awesome. Thanks!